### PR TITLE
Issues 73

### PR DESCRIPTION
**Pull request by WesPainter**
Create BDD tests for put_products

### CODE REVIEW

File: features/steps/shopcart_steps.py [hunk 1]
```diff
@@ -12,15 +12,17 @@
 
 @when(u'I visit the "home page"')
 def step_impl(context):
-    context.resp = context.app.get('/')
+	context.resp = context.app.get('/')
 
 @then(u'I should see "{message}"')
 def step_impl(context, message):
+	print(message)
+	print(context.resp.data)
```
Comment (Pratik-Karnik) [inline features/steps/shopcart_steps.py]: Do we need these print statements?

Comment (WesPainter) [reply features/steps/shopcart_steps.py]: @Pratik-Karnik fixed

File: features/steps/shopcart_steps.py [hunk 2]
```diff
@@ -108,20 +116,34 @@ def step_impl(context,sku,quantity,name,unitprice):
...
-    assert context.resp.status_code == 200
+	context.resp = context.app.get(url)
+	assert context.resp.status_code == 200
+
+@when(u'I change a product with sku "{sku_original}" to sku "{sku_new}", quantity "{quantity}", name "{name}", and unitprice "{unitprice}"')
+def step_impl(context,sku_original, sku_new, quantity, name, unitprice):
+	new_product = { "products": [{"sku" : int(sku_new), "quantity" : int(quantity), "name" : name, "unitprice" : float(unitprice)}] }
+	data = json.dumps(new_product)
+	url = '/shopcarts/'+context.current_shopcart+'/products/' + sku_original
+	context.resp = context.app.put(url, data=data, content_type='application/json')
```
Comment (Pratik-Karnik) [inline features/steps/shopcart_steps.py:142]: I am not sure but should we have asserts for status codes?

File: features/steps/shopcart_steps.py [hunk 3]
```diff
@@ -108,20 +116,34 @@ def step_impl(context,sku,quantity,name,unitprice):
...
+
+@when(u'I change a product with sku "{sku}" to an empty product')
+def step_impl(context,sku):
+	new_product = { "products": [{}] }
+	data = json.dumps(new_product)
+	url = '/shopcarts/'+context.current_shopcart+'/products/' + sku
+	context.resp = context.app.put(url, data=data, content_type='application/json')
```
Comment (Pratik-Karnik) [inline features/steps/shopcart_steps.py:149]: Same here as well. Do we need asserts for status codes?

**Comment by Pratik-Karnik** [comment]
@WesPainter If you feel the asserts arent necessary, the PR seems good to me

**Review by ceciliazhang12 — COMMENTED** [review]
I agree with @Pratik-Karnik, you can add assertions for status code.

**Comment by jiweix** [comment]
I am gonna merge this PR, so we can work on other issues. The assertion for status code can be added later as enhancement

**Comment by WesPainter** [comment]
@Pratik-Karnik @ceciliazhang12 @jiweix I'll add the response code assertions as well. Missed those in the sample code

**Pull request merged**